.github/workflows/test.yml

```diff
@@ -0,0 +1,42 @@
+name: Test Plugin
+
+on:
+  push:
+    branches-ignore:
+      - 'whitesource-remediate/**'
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        test:
+          - test jacocoTestReport
+          - integTest
+          - yamlRestTest
```

Comment:
also just for understanding, what are yamlRestTests for?

Reply:
It's for testing the rest-api-spec [yaml](https://github.com/opensearch-project/OpenSearch/tree/ea1cc9d054b2d7cf0046438ecc57777e29112c2b/rest-api-spec/src/main/resources/rest-api-spec) files. We don't have right now for our plugin, but we would in the future.

Reply:
Here's an example test https://github.com/opensearch-project/OpenSearch/blob/1447d75e5e9ea042dd4bb645d71f25aea1fb42a8/modules/geo/src/yamlRestTest/resources/rest-api-spec/test/geo_shape/10_basic.yml ... it tests a search API with specific input and makes sure it gets the expected output.